client/components/Brands.js

```diff
@@ -0,0 +1,93 @@
+import React from "react";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom";
+import { fetchProducts } from "../store/products";
+
+export class Brands extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      filter: "All",
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+  componentDidMount() {
+    this.props.getProducts();
+  }
+
+  handleChange(evt) {
+    this.setState({ filter: evt.target.value });
+  }
+
+  render() {
+
+    const { products } = this.props;
+    const { filter } = this.state;
+
+    const allProducts = products.filter((product) => {
+      if (filter === "iRobot") {
```

Comment:
This can be greatly simplified by simply having the map return `product.brand === filter`, there's no need to have all of these conditional checks